### PR TITLE
Bug Fix: Init Ctxt Setup Req - Encoding of UE-AMBR

### PR DESCRIPTION
**Pull request by clstefi**
This Bug Fix includes:

1. Length check in UE-AMBR fields. i.e., encode UE_AMBR in 5 Bytes if it is <= UINT32_MAX , else encode the same in 6 Bytes.
2. Clearance of the pre-existing session contexts before allocating a new one, in case of re-attach.

### CODE REVIEW

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

File: src/s1ap/handlers/attach_icsreq.c [hunk 1]
```diff
@@ -189,124 +189,158 @@ icsreq_processing(struct init_ctx_req_Q_msg *g_icsReqInfo)
...
-	 *  0.......         Extension of UEAggregateMaximumBitrate = 0 :Absent
-	 *  .0......         iE-Extensions = 0 :Absent
-	 *                   uEaggregateMaximumBitRateDL(BitRate):
-	 *  ..100...         BitRate Value Range = 4 (offset from 1) :0 to 1099511627775
-	 *  .....000         Padding = 000b
-	 */
-	uint8_t maximum_bit_rate_dl = 0x20;
-	if (g_icsReqInfo->ext_ue_ambr.ext_ue_ambr_DL >= EXT_UEAMBR_MIN
-			|| g_icsReqInfo->ext_ue_ambr.ext_ue_ambr_UL >= EXT_UEAMBR_MIN)
+	if(g_icsReqInfo->exg_max_dl_bitrate <= UINT32_MAX &&
```
Comment (thakurajayL) [inline src/s1ap/handlers/attach_icsreq.c:195]: Comments should have been added here.

File: src/s1ap/handlers/attach_icsreq.c [hunk 2]
```diff
@@ -189,124 +189,158 @@ icsreq_processing(struct init_ctx_req_Q_msg *g_icsReqInfo)
...
+			g_icsReqInfo->exg_max_ul_bitrate <= UINT32_MAX)
+	{
+		datalen = 10;
+
+		maximum_bit_rate_dl = 0x18;
```
Comment (thakurajayL) [inline src/s1ap/handlers/attach_icsreq.c:200]: Would be good how we arrived at these numbers.

**Review by thakurajayL — APPROVED** [review]

**Comment by thakurajayL** [comment]
retest this please

**Pull request merged**